tests/app/forms/handlers/test_number_handler.py

```diff
@@ -105,8 +115,200 @@ def test_percentage_field():
+        'label': 'Max Test',
+        'mandatory': False,
+        'validation': {
+            'messages': {
+                'INVALID_NUMBER': 'Please only enter whole numbers into the field.',
+                'NUMBER_TOO_LARGE': 'The maximum value allowed is 20. Please correct your answer.',
+            }
+        },
+        'id': 'test-range',
+        'type': 'Currency',
+    }
+
+    test_form_class = get_test_form_class(answer_schema)
+    form = test_form_class(DummyPostData(test_field=['21']))
+
+    form.validate()
+
+    assert (
+        form.errors['test_field'][0]
+        == answer_schema['validation']['messages']['NUMBER_TOO_LARGE']
+    )
+
+
+def test_zero_max(app):
+    max_value = 0
+
+    answer_schema = {
+        'max_value': {'value': max_value},
+        'label': 'Max Test',
+        'mandatory': False,
+        'id': 'test-range',
+        'type': 'Currency',
+    }
+    error_message = error_messages['NUMBER_TOO_LARGE'] % dict(max=max_value)
+
+    test_form_class = get_test_form_class(answer_schema, messages=error_messages)
+    form = test_form_class(DummyPostData(test_field=['1']))
+    form.validate()
+
+    assert form.errors['test_field'][0] == error_message
+
+
+def test_zero_min(app):
+    min_value = 0
+
+    answer_schema = {
+        'max_value': {'value': min_value},
+        'label': 'Max Test',
+        'mandatory': False,
+        'id': 'test-range',
+        'type': 'Currency',
+    }
+    error_message = error_messages['NUMBER_TOO_SMALL'] % dict(min=min_value)
+
+    test_form_class = get_test_form_class(answer_schema, messages=error_messages)
+    form = test_form_class(DummyPostData(test_field=['-1']))
+    form.validate()
+
+    assert form.errors['test_field'][0] == error_message
+
+
+def test_value_range(app):
+    answer_schema = {
+        'min_value': {'value': 10},
+        'max_value': {'value': 20},
+        'label': 'Range Test 10 to 20',
+        'mandatory': False,
+        'validation': {
+            'messages': {
+                'INVALID_NUMBER': 'Please only enter whole numbers into the field.',
+                'NUMBER_TOO_SMALL': 'The minimum value allowed is 10. Please correct your answer.',
+                'NUMBER_TOO_LARGE': 'The maximum value allowed is 20. Please correct your answer.',
+            }
+        },
+        'id': 'test-range',
+        'type': 'Currency',
+    }
+
+    test_form_class = get_test_form_class(answer_schema)
+    form = test_form_class(DummyPostData(test_field=['9']))
+    form.validate()
+
+    assert (
+        form.errors['test_field'][0]
+        == answer_schema['validation']['messages']['NUMBER_TOO_SMALL']
+    )
+
+
+def test_manual_min_exclusive(app):
+    answer_schema = {
+        'min_value': {'value': 10, 'exclusive': True},
+        'label': 'Min Test',
+        'mandatory': False,
+        'validation': {
+            'messages': {
+                'INVALID_NUMBER': 'Please only enter whole numbers into the field.',
```

Comment:
I don't think `INVALID_NUMBER` is used in the test and should be removed. This applies to other tests with validation messages that aren't used.